### PR TITLE
Year In Music Initial setup

### PR DESCRIPTION
**Pull request by 07jasjeet**
### Tasks Completed :

1) Added Yim api and its data classes.
2) Hardcoded some values in data class which are guaranteed to get data. This helps avoiding unnecessary null checks.
2) Tested all anomalies and left comments in code for reference.
3) Created YimModule for dependency injection.
4) Created YimViewModel and YimRepository.
5) Implemented login check and internet observer and checks.
6) Home Screen Implemented.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -74,6 +77,18 @@ android {
         kotlinCompilerExtensionVersion '1.3.0'
     }
     namespace 'org.listenbrainz.android'
+    compileOptions {
```
Comment (akshaaatt) [inline app/build.gradle:80]: We don't need these additions. Please remove

File: app/build.gradle [hunk 2]
```diff
@@ -100,6 +115,8 @@ dependencies {
 
     //Image downloading and Caching library
     implementation 'com.github.bumptech.glide:glide:4.14.2'
+    implementation 'androidx.activity:activity-compose:1.3.1'
+    debugImplementation "androidx.compose.ui:ui-test-manifest:$compose_version"
```
Comment (akshaaatt) [inline app/build.gradle:119]: Please move this to the tests section down

File: app/src/test/java/org/listenbrainz/android/MockLookupRepository.kt [hunk 3]
```diff
@@ -37,7 +37,7 @@ class MockLookupRepository : LookupRepository {
         TODO("Not yet implemented")
     }
 
-    override suspend fun fetchAcoustIDResults(duration: Long, fingerprint: String?): Resource<List<Recording>> {
+    suspend fun fetchAcoustIDResults(duration: Long, fingerprint: String?): Resource<List<Recording>> {
```
Comment (akshaaatt) [inline app/src/test/java/org/listenbrainz/android/MockLookupRepository.kt:40]: Please remove this. I think we missed this while the cleanup was being done

**Review by akshaaatt — APPROVED** [review]
Awesome work! 💯 Looks crisp and clean! Thank you so much for this amazing work. Let's get this merged once the comments left are taken care of and I test this on a device once.

File: app/src/main/java/org/listenbrainz/android/presentation/features/yim/navigation/YimNavigation.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import androidx.compose.runtime.Composable
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import org.listenbrainz.android.presentation.features.yim.YearInMusicActivity
+import org.listenbrainz.android.presentation.features.yim.YimViewModel
+import org.listenbrainz.android.presentation.features.yim.screens.YimHomeScreen
+import org.listenbrainz.android.presentation.features.yim.screens.YimMainScreen
+
+@RequiresApi(Build.VERSION_CODES.N)
```
Comment (akshaaatt) [inline app/src/main/java/org/listenbrainz/android/presentation/features/yim/navigation/YimNavigation.kt:14]: Avoid this at all times. We shouldn't be in favor of this while releasing a feature. Either we bump the minimum Android version to 24 or we look for alternatives. For now, I think we can bump the minimum Android version to 24

Comment (07jasjeet) [reply app/src/main/java/org/listenbrainz/android/presentation/features/yim/navigation/YimNavigation.kt:14]: I had an alternative but this makes life so much easier for all features. Internet observer has been added to util with proper documentation.

File: app/src/test/java/org/listenbrainz/android/MockLookupRepository.kt (hunk 3 above)
Comment (07jasjeet) [reply app/src/test/java/org/listenbrainz/android/MockLookupRepository.kt:40]: Done

File: app/build.gradle (hunk 2 above)
Comment (07jasjeet) [reply app/build.gradle:119]: Sorted

File: app/build.gradle (hunk 1 above)
Comment (07jasjeet) [reply app/build.gradle:80]: Done

File: app/src/main/java/org/listenbrainz/android/presentation/features/dashboard/DashboardBackdropBackLayerContent.kt [hunk 5]
```diff
@@ -122,6 +125,26 @@ fun BackLayerContent(activity: Activity, applicationContext: Context) {
...
+                            Intent(
+                                applicationContext,
+                                YearInMusicActivity::class.java
+                            )
+                        )
+                    }),
+                elevation = 0.dp,
+                backgroundColor = MaterialTheme.colors.onSurface
+            ){
+                Text(text = "Yim Activity Test", color = Color.White, modifier = Modifier.padding(16.dp))
```
Comment (akshaaatt) [inline app/src/main/java/org/listenbrainz/android/presentation/features/dashboard/DashboardBackdropBackLayerContent.kt:146]: Can we fix this in this PR itself?

Comment (akshaaatt) [reply app/src/main/java/org/listenbrainz/android/presentation/features/dashboard/DashboardBackdropBackLayerContent.kt:146]: "Your Year in Music" could be the title and the usual logo could be beside it

Comment (07jasjeet) [reply app/src/main/java/org/listenbrainz/android/presentation/features/dashboard/DashboardBackdropBackLayerContent.kt:146]: My bad, forgot about this one. I was just waiting for aerozol to get me some logo to display on the card.

Comment (akshaaatt) [reply app/src/main/java/org/listenbrainz/android/presentation/features/dashboard/DashboardBackdropBackLayerContent.kt:146]: Let's do the official YIM logo in the next PR. For now, a placeholder is fine

Comment (07jasjeet) [reply app/src/main/java/org/listenbrainz/android/presentation/features/dashboard/DashboardBackdropBackLayerContent.kt:146]: Alright!

**Review by akshaaatt — APPROVED** [review]
Perfect! Thanks a lot 💯

**Pull request merged**